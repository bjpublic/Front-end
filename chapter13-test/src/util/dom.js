export function addClass(el, ...classNames) {
  el.classList.add(...classNames);
}
