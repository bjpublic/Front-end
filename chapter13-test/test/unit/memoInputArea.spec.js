import { screen } from "@testing-library/dom";
import userEvent from "@testing-library/user-event";
import MemoInputArea from "../../src/memoInputArea";

describe("memoInputArea", () => {
  let memoInputArea, el, spy;

  beforeEach(() => {
    spy = jest.fn();
    el = document.createElement("div");
    memoInputArea = new MemoInputArea(el, { addMemoItem: spy });

    document.body.append(el);
  });

  afterEach(() => {
    el.remove();
  });

  it("MemoInputArea 요소가 렌더링된다.", () => {
    expect(el).toContainElement(memoInputArea.inputAreaEl);
  });

  it("MemoInputArea 요소는 스냅샷과 일치해야 한다.", () => {
    expect(memoInputArea.inputAreaEl).toMatchSnapshot();
  });

  it("Add Memo 버튼을 누를 시 입력된 텍스트 값을 인자로 받는 addMemoItem() 함수가 호출된다.", () => {
    const button = screen.getByText("Add Memo");
    const textarea = screen.getByLabelText("메모 작성하기");

    userEvent.type(textarea, "Bob");
    userEvent.click(button);

    expect(spy).toHaveBeenCalledWith("Bob");
    expect(textarea).toHaveValue("");
  });
});
