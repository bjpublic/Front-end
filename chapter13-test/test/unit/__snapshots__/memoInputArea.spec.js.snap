// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`memoInputArea MemoInputArea 요소는 스냅샷과 일치해야 한다. 1`] = `
<div
  class="border-bottom mb-3 pb-3"
>
  <div
    class="form-floating"
  >
    <textarea
      class="form-control"
      id="memo"
      style="height: 100px;"
    />
    <label
      for="memo"
    >
      메모 작성하기
    </label>
  </div>
  <div
    class="d-flex justify-content-end mt-3"
  >
    <button
      class="btn btn-primary"
    >
      Add Memo
    </button>
  </div>
</div>
`;
